### PR TITLE
Add caseworker-upload-documents-and-submit event

### PR DESCRIPTION
**Pull request by mapingo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-731
https://tools.hmcts.net/jira/browse/NFDIV-591

### Change description ###
Add caseworker-upload-documents-and-submit event to allow caseworker to upload documents and submit case

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerUploadDocumentsAndSubmitTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    private String awaitingDocumentsStateResponse() throws IOException {
+        return expectedResponse("classpath:caseworker-document-upload-about-to-submit-response.json")
+            .replace("<state>", AwaitingDocuments.getName());
+    }
+
+    private String submittedStateResponse() throws IOException {
+        return expectedResponse("classpath:caseworker-document-upload-about-to-submit-response.json")
+            .replace("<state>", Submitted.getName());
+    }
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerUploadDocumentsAndSubmitTest.java:122]: can be one method if you pass state as the argument

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorCreateApplication.java [hunk 2]
```diff
@@ -109,8 +109,8 @@ private PageBuilder addEventConfig(
             .endButtonLabel("Save Application")
             .aboutToSubmitCallback(this::aboutToSubmit)
             .explicitGrants()
-            .grant(CREATE_READ_UPDATE, CASEWORKER_DIVORCE_SOLICITOR)
+            .grant(CREATE_READ_UPDATE, CASEWORKER_DIVORCE_SOLICITOR, CASEWORKER_DIVORCE_COURTADMIN, CASEWORKER_DIVORCE_COURTADMIN_BETA)
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorCreateApplication.java:112]: do we need court admin role to have CRD for solicitor create ?

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorUpdateApplication.java [hunk 3]
```diff
@@ -108,11 +108,8 @@ private PageBuilder addEventConfig(final ConfigBuilder<CaseData, State, UserRole
             .showSummary()
             .aboutToSubmitCallback(this::aboutToSubmit)
             .explicitGrants()
-            .grant(CREATE_READ_UPDATE, CASEWORKER_DIVORCE_SOLICITOR)
+            .grant(CREATE_READ_UPDATE, CASEWORKER_DIVORCE_SOLICITOR, CASEWORKER_DIVORCE_COURTADMIN, CASEWORKER_DIVORCE_COURTADMIN_BETA)
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorUpdateApplication.java:111]: same for this one

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorCreateApplication.java (hunk 2 above)
Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorCreateApplication.java:112]: This is a work around, until we have a fix for the generator authorisation,  if I don't add caseworker access here then the caseworker is not able to upload documents.  Ideally we need a finer grain control over the authorisation on single fields.

**Review by nitinprabhu — APPROVED** [review]

**Pull request closed without merging**